ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java

```diff
@@ -33,462 +32,109 @@
+    protected String                     title;
+    protected String                     message;
+    protected ReadState                  state;
+    protected ProjectConfigDto           project;
+
+    private   List<NotificationObserver> observers;
+
+    protected DelayedTask setUnreadStateTask = new DelayedTask() {
+        @Override
+        public void onExecute() {
+            time = System.currentTimeMillis();
+            setState(UNREAD);
+        }
+    };
 
-    /**
-     * Create notification with message, type, note about important this one and action delegates on opening and closing of a
-     * notification. Other parameters will be added with default values. This notification have got an unread state, a finished status.
-     * This notification delegates some actions in response to opening and closing of a notification.
-     *
-     * @param message
-     *         notification's message
-     * @param type
-     *         notification's type
-     * @param important
-     *         note about important this notification
-     * @param openHandler
-     *         delegate that provides some actions when opening notification
-     * @param closeHandler
-     *         delegate that provides some actions when closing notification
-     */
-    public Notification(@NotNull String message, @NotNull Type type, boolean important,
-                        @Nullable OpenNotificationHandler openHandler, @Nullable CloseNotificationHandler closeHandler) {
-        this(message, type, FINISHED, UNREAD, new Date(), important, openHandler, closeHandler);
+    public Notification(@NotNull String title,
+                        @Nullable String message) {
+        this(null, title, message);
     }
 
-    /**
-     * Create notification with message, type, status, note about important this one and action delegates on opening and closing of a
-     * notification. Other parameters will be added with default values. This notification have got an unread state. This notification
-     * delegates some actions in response to opening and closing of a notification.
-     *
-     * @param message
-     *         notification's message
-     * @param type
-     *         notification's type
-     * @param status
-     *         notification's status
-     * @param important
-     *         note about important this notification
-     * @param openHandler
-     *         delegate that provides some actions when opening notification
-     * @param closeHandler
-     *         delegate that provides some actions when closing notification
-     */
-    public Notification(@NotNull String message, @NotNull Type type, @NotNull Status status, boolean important,
-                        @Nullable OpenNotificationHandler openHandler, @Nullable CloseNotificationHandler closeHandler) {
-        this(message, type, status, UNREAD, new Date(), important, openHandler, closeHandler);
+    public Notification(@Nullable ProjectConfigDto project,
```

Comment:
It would be better to put this nullable (optional) parameter to the end of parameters list